pacsman/pynetdicom_client.py

```diff
@@ -122,61 +120,115 @@ def studies_for_patient(self, patient_id, additional_tags=None):
 
             return datasets
 
+    def search_series(self, query_dataset, additional_tags=None):
+        additional_tags = additional_tags or []
+        query_dataset.QueryRetrieveLevel = 'INSTANCE'
+        additional_tags += [
+            'Modality',
+            'BodyPartExamined',
+            'SeriesDescription',
+            'SeriesDate',
+            'SeriesTime',
+            'PatientPosition',
+        ]
+        add_missing_blank_tags_to_dataset(query_dataset, additional_tags)
+        ae = AE(ae_title=self.client_ae, scu_sop_class=QueryRetrieveSOPClassList)
+
+        datasets = []
+        with association(ae, self.pacs_url, self.pacs_port) as assoc:
+            responses = assoc.send_c_find(query_dataset, query_model='S')
+            for series in checked_responses(responses):
+                if hasattr(series, 'SeriesInstanceUID'):
+                    datasets.append(series)
+        return datasets
+
     def series_for_study(self, study_id, modality_filter=None, additional_tags=None):
+        additional_tags = additional_tags or []
 
         ae = AE(ae_title=self.client_ae, scu_sop_class=QueryRetrieveSOPClassList)
 
         with association(ae, self.pacs_url, self.pacs_port) as assoc:
             dataset = Dataset()
             dataset.StudyInstanceUID = study_id
+            dataset.QueryRetrieveLevel = 'SERIES'
 
+            additional_tags += [
+                'SeriesInstanceUID',
+                'BodyPartExamined',
+                'SeriesDescription',
+                'SeriesDate',
+                'SeriesTime',
+                'PatientPosition',
+                'NumberOfSeriesRelatedInstances',
+            ]
+            add_missing_blank_tags_to_dataset(dataset, additional_tags)
```

Comment:
I really like the new util functions. This one is probably better named `add_blank_tags_to_dataset` instead of `add_missing_blank_tags_to_dataset`, because in this case and others it's adding blank tags for the search query, not tags the caller expects that are missing from the source dataset.

Reply:
If there is already a tag in the query, it won't add it. So maybe `set_undefined_tags_to_blank` would be more clear?

Reply:
👍